src/fights.ts

```diff
@@ -1400,39 +1402,89 @@ const freeFightSources = [
-      if (get("_shadowAffinityToday")) return false;
+      if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) return true;
 
-      if (!ClosedCircuitPayphone.rufusTarget()) return true;
       if (get("rufusQuestType") === "items") {
         return false; // We deemed it unprofitable to complete the quest in potionSetup
       }
       if (get("encountersUntilSRChoice") === 0) {
         // Target is either an artifact or a boss
         return true; // Get the artifact or kill the boss immediately for free
       }
-      return false; // We have to spend turns to get the artifact or kill the boss
+
+      // Consider forcing noncombats below:
+      if (globalOptions.prefs.yachtzeechain) return false; // NCs are better when yachtzeeing, probably
+      // TODO: With the KoL update, is there a function for checking if an NC is already forced?
+      if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
+        return true;
+      }
+
+      // TODO: Calculate this against using the +5 fam weight buff
+      if (have($item`Cincho de Mayo`) && useableCinch() >= 60) {
+        return true;
+      }
+      return false; // It costs turns to do anything else here
     },
     () => {
-      if (have($item`Rufus's shadow lodestone`)) setChoice(1500, 2);
+      if (have($item`Rufus's shadow lodestone`)) {
+        setChoice(1500, 2); // Turn in lodestone if you have it
+        adv1(bestShadowRift(), -1, "");
+      }
       if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) {
         ClosedCircuitPayphone.chooseQuest(() => 2); // Choose an artifact (not supporting boss for now)
       }
-      adv1(bestShadowRift(), -1, "");
 
-      if (get("encountersUntilSRChoice", 0) === 0) {
-        if (ClosedCircuitPayphone.have() && !ClosedCircuitPayphone.rufusTarget()) {
-          ClosedCircuitPayphone.chooseQuest(() => 2);
+      // Potentially force an NC
+      let ncForced = get("encountersUntilSRChoice", 1) === 0;
+      if (!ncForced && !globalOptions.prefs.yachtzeechain && get("rufusQuestType") !== "items") {
+        if (have($item`Clara's bell`) && !globalOptions.clarasBellClaimed) {
+          globalOptions.clarasBellClaimed = true;
+          use($item`Clara's bell`);
+          ncForced = true;
+          // Jurassic parka
+        } else if (
+          have($item`Jurassic Parka`) &&
+          get("_spikolodonSpikeUses") < 5 &&
+          have($effect`Shadow Affinity`)
+        ) {
+          freeFightOutfit(new Requirement([], { forceEquip: $items`Jurassic Parka` }));
+          cliExecute("parka spikolodon");
+          const macro = Macro.skill($skill`Launch spikolodon spikes`).basicCombat();
+          // Adventure until we use spikes
+          const startingSpikes = get("_spikolodonSpikeUses");
+          do {
+            garboAdventureAuto(bestShadowRift(), macro);
+          } while (get("_spikolodonSpikeUses") === startingSpikes);
```

Comment:
I don't like this `do... while` loop. If something goes wrong (e.g. misalignment of the prefs) we could potentially burn every single turn here.

Imo, this should be a single `garboAdventureAuto()` call, and we should throw if we did not manage to spike like we intended.

Reply:
This is probably OOS of this PR, but we shouldn't use spikes (or any other NC forcer, for that matter) if you only have a few turns left before the NC; we should use them when your distance from it is maximal.

But spikes in particular, if you use them with 1 turn left you've achieved nothing.